### ISSUE TITLE
missing module DataSet  

### ISSUE THREAD

**Issue by sarala777** [iamownt/LSTM-DeepFM#1]
In the code , the module DataSet  is missing to import this line "from dataset.DataSet import PretrainDataset".
could you pls update it.

**Comment by iamownt** [comment]
Thanks for your question, we'll reorganize the code and fix this in two days.

**Comment by iamownt** [comment]
1. We reconstructed the code base in order to make it more pythonic. 
2. We discovered that, in  LSTM-DeepFM/pretraining/selfsupervised_pretraining.py, row 94, the std as denominator may cause excessive losses, the std is dataset dependent.
3. We discovered that the scipy optimize can further optimize the performance of the model. FM is great useful if there is many discrete features.、
```
from scipy.optimize import minimize
from sklearn.metrics import mean_squared_error
def f(x):
   pred1 Deep  pred2 FM
   pred = x[0]*pred1 + x[1]*pred2
   score = mean_squared_error(pred, label)
    return score

weight_init = [0.5, 0.5]
result = minimize(f, weight_init , method="Nelder-Mead")
print(result.x)
```
4. We recommend the https://github.com/JustusvLiebig/Soft_Sensor_Experiments, since there is many implemented soft sensor models.
